textpipe/doc.py

```diff
@@ -608,3 +608,33 @@ def summary(self):
         []
         """
         return self.generate_textrank_summary()
+        
+                
+    @functools.lru_cache()
+    def extract_lead(self, n=3):
+        """
+        returns the lead-N sentences (text only) of the document 
+        if the text is smaller than the requested N, return full text
+        """
+        if n < len(self.sents):
+            return [s[0] for s in self.sents[:n]]
+        return [s[0] for s in self.sents]
```

Comment:
I don't think the `if` statement is needed.

You can just do this unconditionally: 
```python
return [s[0] for s in self.sents[:n]]
```

Reply:
true :)

Reply:
or not actually, because then it can return an index out of bounds error...

Reply:
```python
>>> a = list(range(10))
>>> a[:20]
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
```